src/js/controllers/imageCtrl-test.js

```diff
@@ -0,0 +1,130 @@
+                    ispublic: 1,
+                    owner: "138698049@N03",
+                    secret: "1d4dc255d0",
+                    server: "693",
+                    title: ""
+                }
+            ]
+        }
+
+    };
+    mockImageArray = ['https://farm6.staticflickr.com/5764/23780914905_0eb2a77a55_z.jpg', 'https://farm1.staticflickr.com/693/23698470321_1d4dc255d0_z.jpg'];
+
+    beforeEach(module('flickrPOC'));
+
+    describe('imagesCtrl', function(){
+
+        describe('Sucessful call to api', function() {
+            beforeEach(module(function ($provide) {
+                $provide.service('images', function ($q) {
+                    this.loadPics = sinon.stub().returns($q.resolve(sampleJSON));
+                    this.createImagesArray = sinon.stub().returns($q.resolve(mockImageArray));
+                });
+
+            }));
+
+            beforeEach(inject(function (_$rootScope_, _$controller_, _images_, _$q_, _$log_) {
+                $rootScope = _$rootScope_.$new();
+                $controller = _$controller_;
+                images = _images_;
+                $q = _$q_;
+                $log = _$log_;
+                vm = $controller('imagesCtrl');
+
+                $rootScope.$apply();
+
+            }));
+
+            describe('return flickr images from object from http call', function(){
+                var imageArray;
+
+                beforeEach(function(){
+                    imageArray = vm.images.$$state.value;
+                });
+
+               it('should return an array of images', function(){
+                   expect(angular.isArray(imageArray)).to.equal(true);
+               });
+
+               it('Should have all strings', function(){
+                   angular.forEach(imageArray, function(value){
+                       expect(angular.isString(value)).to.equal(true);
+                   });
+               });
+
+               it('Each strings must have an http address', function(){
+                   angular.forEach(imageArray, function(value){
+                       expect(value.substr(0, 8)).to.equal('https://');
+                   });
+               });
+
+               it('Each strings must have an flickr address', function(){
+                   angular.forEach(imageArray, function(value){
+                       expect(value).to.contain('.staticflickr.com');
+                   });
+               });
+
+               it('Each strings must have an flickr address', function(){
+                   angular.forEach(imageArray, function(value){
+                       expect(value.substr(-4, 4)).to.match(/^\.|\.jpg$|\.gif$|.png$/);
+                   });
+               });
+
+            });
+
+        });
+
+        describe('Error from to api', function() {
+
+            beforeEach(module(function($provide){
+                $provide.service('images', function($q, $log){
+                    this.loadPics = sinon.stub().returns($q.reject($log));
```

Comment:
This doesn't look right. The loadPics method wouldn't be returning the $log service, but an error object of sorts. The $log service should be used to log the message, but the return value from the loadPics method itself should be an object.